lib/CL/clCreateProgramWithBinary.c

```diff
@@ -123,14 +124,51 @@ POname(clCreateProgramWithBinary)(cl_context                     context,
   program->kernels = NULL;
   program->build_status = CL_BUILD_NONE;
 
+  /* We do no want a binary with different kind of binary in it
```

Comment:
Please add a section to the user manual that describes the different supported binary input formats.